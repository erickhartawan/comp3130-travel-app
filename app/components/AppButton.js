import React from 'react';
import { Text, TouchableOpacity, StyleSheet, View} from 'react-native';


import BaseColors from '../config/BaseColors';


function AppButton({title, color="primary", onPress}) {
    return (
        <TouchableOpacity onPress={onPress}>
            <View style={[styles.button,{backgroundColor:BaseColors[color]}]}>
                <Text style={styles.text}> {title}  </Text>
            </View>
        </TouchableOpacity>

    );
}

const styles = StyleSheet.create({
    button:{
        backgroundColor: BaseColors.primary,
        borderRadius: 20,
        width: '100%',
        padding: 15,
        justifyContent: 'center',
        alignItems: 'center',
    },
    text:{
        color: BaseColors.tertiary,
        fontSize: 16,
        textTransform: 'uppercase',
        fontWeight: 'bold',
    },
})

export default AppButton;
