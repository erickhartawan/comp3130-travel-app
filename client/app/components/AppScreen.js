import React from 'react';
import { View, SafeAreaView, StyleSheet,Text } from 'react-native';
import Constants from 'expo-constants';
import BaseColors from '../config/BaseColors';


function AppScreen({children, style}) {
    return (
        <SafeAreaView style={[styles.screen,style]}>
            <View style={styles.paddingView}>
                {children}
            </View>
        </SafeAreaView>

    );
}

const styles = StyleSheet.create({
    screen:{
        flex:1,
        // marginTop: Constants.statusBarHeight,
        alignSelf:'stretch',
        padding: 20,
        width:'100%',
        backgroundColor: BaseColors.background,
        overflow: 'visible'
    },
    paddingView:{
        flex:1,
    }
})

export default AppScreen;
