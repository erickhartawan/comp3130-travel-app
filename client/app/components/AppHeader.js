import React from 'react'
import { View, Text,StyleSheet } from 'react-native'
import { useNavigation } from '@react-navigation/core'

import BaseColors from '../config/BaseColors'
import AppButton from './AppButton'

const AppHeader = () => {
    const navigation = useNavigation();
    return (
        <View style={styles.container}> 
            <AppButton textStyle={styles.textStyle} style={styles.headerButton} title="MyList" onPress={() => navigation.navigate("MyList")} />
            <AppButton textStyle={styles.textStyle} style={styles.headerButton} title="" />
            {/* <AppButton textStyle={styles.textStyle} style={styles.headerButton} title='Add Listing' onPress={() =>  } /> */}
        </View>
    )
}

const styles = StyleSheet.create({
    container:{
        flex: 1,
        flexDirection:'row',
        justifyContent: 'space-between',
        padding: 5
    },
    headerButton:{
        backgroundColor:'#FFF'
    },

    textStyle:{
        color: BaseColors.primary
    }
})

export default AppHeader
