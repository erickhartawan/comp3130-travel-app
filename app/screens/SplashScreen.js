import React from 'react'
import { View, Text,Image,StyleSheet} from 'react-native'
import AppIcon from '../components/AppIcon'
import AppScreen from '../components/AppScreen'
import BaseColors from '../config/BaseColors'
const SplashScreen = ({navigation}) => {
    setTimeout(() => {
        navigation.navigate("Login")
    }, 3000);
    return (
        <AppScreen>
            <Image 
            style={styles.logo}
            source={require('./../../assets/logoCorrect.png')}
            />
            <Text style={styles.version}>
                v.0.0.1
            </Text>
            <Text style={styles.message}>
                You will be redirected to Register page in 3s
            </Text>
        </AppScreen>
    )
}

const styles = StyleSheet.create({
    logo:{
        flex:2,
        width:'100%',
        resizeMode: 'contain'
    },
    version:{
        flex:2,
        alignSelf:'center',
        color: BaseColors.secondary
    },
    message:{
        flex:1,
        alignSelf:'center',
        color: BaseColors.primary 
    }
})


export default SplashScreen