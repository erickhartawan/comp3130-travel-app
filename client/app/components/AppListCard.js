import React from 'react'
import { View, Pressable , StyleSheet, Image} from 'react-native'
import Swipeable from 'react-native-gesture-handler/Swipeable';
import BaseColors from '../config/BaseColors';
import AppButton from './AppAbsoluteButton';
import AppScreen from './AppScreen';

import AppText from './AppText'

const AppListCard = ({title,subtitle,rating,handlePress,description,onPress, onDelete,onSwipeLeft, onSwipeRight,image}) => {
    
    return (
    <Swipeable
        >
        <Pressable sytle={styles.pressableContainer} onPress={handlePress}>
            <View style={styles.container}>
                <View style={styles.innerContainer}>
                    <Image source={{uri: `${image}`}} style={styles.image} />
                    <AppScreen style={styles.textContainer}>
                        <AppText style={styles.title}>{title}</AppText>
                        <AppText style={styles.subtitle}>{subtitle}</AppText>
                        <AppText style={styles.rating}>{`${rating} out of 10`}</AppText>
                    </AppScreen >
                </View>
                <View style={styles.buttonContainer}>
                    <AppButton title='Edit' color={BaseColors.tertiary}/>
                    <AppButton title='Remove' color={BaseColors.tertiary} onPress={()=> onDelete()}/>
                </View>
            </View>
            

        </Pressable>

    </Swipeable>
    )
}

const styles = StyleSheet.create({
    container:{
        flexDirection: 'column',
        backgroundColor: BaseColors.secondary,
        color: BaseColors.text,
        padding: 7,
        borderStyle: 'solid',
        borderWidth: 2,
        borderColor: BaseColors.secondary,
        borderRadius: 10,
        margin: 10,
    },
    innerContainer:{
        flexDirection: 'row',
        backgroundColor: BaseColors.secondary,
        color: BaseColors.text,
        flex:1
    },
    buttonContainer:{
        alignContent:'center',
        justifyContent:'center',
        padding:5,
        flexDirection: 'row'
    },
    textContainer:{
        flexDirection:'column',
        backgroundColor:BaseColors.secondary,
        padding:5,
    },
    title:{
        flex: 2,
        color: BaseColors.text,
        flexShrink:1

    },
    subtitle:{
        flex:1,
        color: BaseColors.text,
        flexShrink:1
    },
    rating:{
        flex: 1,
        color: BaseColors.text,
        flexShrink:1
    },
    pressableContainer:{
        backgroundColor: BaseColors.secondary,
        color: BaseColors.text,
        flex: 1,
    },
    image:{
        width:100,
        height:100,
        resizeMode: 'contain'
    }
})


export default AppListCard
