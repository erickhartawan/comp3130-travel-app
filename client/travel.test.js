import React from 'react'
import renderer from 'react-test-renderer';
import AppButton from './app/components/AppButton';
import AppHeader from './app/components/AppHeader';
import AppListCard from './app/components/AppListCard';
import AppScreen from './app/components/AppScreen';
import AppText from './app/components/AppText';
import SplashScreen from './app/screens/SplashScreen';

it('Button renders correctly', () =>{
    const tree = renderer
    .create(<AppButton
                style={{ backgroundColor:"#4B77BE"}}
                onPress={()=> console.log("Button pressed")}    
    />)
    .toJSON();
    expect(tree).toMatchSnapshot();
})
it('Screen renders correctly', () =>{
    const tree = renderer
    .create(<AppScreen
                style={{flex:1,alignSelf:'stretch',padding: 20,width:'100%',}}
    />)
    .toJSON();
    expect(tree).toMatchSnapshot();
})

it('Text renders correctly', () =>{
    const tree = renderer
    .create(<AppText
    />)
    .toJSON();
    expect(tree).toMatchSnapshot();
})

it('Splash Screen renders correctly', () =>{
    const tree = renderer
    .create(<SplashScreen
    
    />)
    .toJSON();
    expect(tree).toMatchSnapshot();
})
