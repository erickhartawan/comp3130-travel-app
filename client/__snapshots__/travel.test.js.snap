// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
>
  <View
    accessible={true}
    style={
      Object {
        "backgroundColor": "#4B77BE",
        "marginBottom": 10,
        "opacity": 1,
        "zIndex": 18,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#581FFF",
            "borderRadius": 20,
            "justifyContent": "center",
            "marginBottom": 10,
            "padding": 15,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#581FFF",
          },
          Object {
            "backgroundColor": "#4B77BE",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFAFA",
              "fontSize": 16,
              "fontWeight": "bold",
              "textTransform": "uppercase",
            },
            undefined,
          ]
        }
      >
         
          
      </Text>
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`Screen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#282828",
        "flex": 1,
        "overflow": "visible",
        "padding": 20,
        "width": "100%",
      },
      Object {
        "alignSelf": "stretch",
        "flex": 1,
        "padding": 20,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
</RCTSafeAreaView>
`;

exports[`Splash Screen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#282828",
        "flex": 1,
        "overflow": "visible",
        "padding": 20,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "flex": 4,
          "resizeMode": "contain",
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "flexShrink": 1,
            "fontFamily": "Avenir-Roman",
            "fontSize": 20,
            "justifyContent": "center",
          },
          Object {
            "alignSelf": "center",
            "color": "#FFFAFA",
            "flex": 2,
            "padding": 20,
          },
        ]
      }
    >
       
      v.0.0.1
       
    </Text>
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "flexShrink": 1,
            "fontFamily": "Avenir-Roman",
            "fontSize": 20,
            "justifyContent": "center",
          },
          Object {
            "alignSelf": "center",
            "color": "#FFFAFA",
            "flex": 1,
            "flexShrink": 2,
            "padding": 20,
          },
        ]
      }
    >
       
      You will be redirected to Register page in 2s
       
    </Text>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "marginBottom": 10,
            "opacity": 1,
            "zIndex": 18,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#581FFF",
                "borderRadius": 20,
                "justifyContent": "center",
                "marginBottom": 10,
                "padding": 15,
                "width": "100%",
              },
              Object {
                "backgroundColor": "#581FFF",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFAFA",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "textTransform": "uppercase",
                },
                undefined,
              ]
            }
          >
             
            not redirected? click here instead
              
          </Text>
        </View>
      </View>
    </RNGestureHandlerButton>
  </View>
</RCTSafeAreaView>
`;

exports[`Text renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "flex": 1,
        "flexShrink": 1,
        "fontFamily": "Avenir-Roman",
        "fontSize": 20,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
   
   
</Text>
`;
